[AY1920S1-CS2113-T16-1] Le Duc

laurenthg opened the pull request:
@rshah918
@laurenthg
@huangje

rshah918 committed:
Merge pull request #76 from huangje/master

Show by option

laurenthg committed:
Merge pull request #79 from rshah918/master

Statistics

laurenthg committed:
fix small bug: crash when command is "find"

laurenthg committed:
Pass the text-ui-test

laurenthg committed:
Documentation: Update UG

laurenthg committed:
Fix crash for "reschedule"

laurenthg committed:
javadoc throw in FindCommand

rshah918 committed:
Merge pull request #80 from laurenthg/debug

Debug

rshah918 committed:
Merge pull request #81 from laurenthg/documentation

Documentation: Update UG

Commit:
shortcut respect the multi step constraint

laurenthg committed:
Merge pull request #82 from huangje/master

shortcut respect the multi step constraint

laurenthg committed:
v1.2 jar

rshah918 committed:
Merge pull request #83 from laurenthg/master

v1.2 jar

huangje committed:
add helper method for showCommand

And add junit test

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
hotfix shortcut

in the parser, we couldn't use the default command for certain command

laurenthg committed:
Merge pull request #85 from huangje/master

Clean the showCommand code a bit

rshah918 committed:
Merge pull request #3 from AY1920S1-CS2113-T16-1/master

update fork

sanjukta99 commented on src/main/java/leduc/Date.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    /**
+     * Setter of the LocalDateTime d
```
JavaDoc comments should be of the form 
`Sets the local date and time of ... `
`Returns.... `
`@param ...`
You guys can take a look at the Java Coding Standards on the module website and update the JavaDoc comments

sanjukta99 commented on src/main/java/leduc/Parser.java:
```diff
@@ -0,0 +1,143 @@
+        else if (user.matches(ByeCommand.getByeShortcut()) || user.matches("bye")){
+            c = new ByeCommand(user);
+        }
+
```
Perhaps a switch case may make this easier to read?

sanjukta99 commented on src/main/java/leduc/Ui.java:
```diff
@@ -0,0 +1,182 @@
+    public void showNotCompleteList(ArrayList<Task> notCompleteTasks, TaskList tasks){//print the task so they have the same index
+        System.out.println("\t---------------------------------------------------------------------------------");
+        System.out.println("\t Here are the tasks in your list:");
+        for(int i = 0; i < tasks.size(); i++){
```
I'm guessing this message was supposed to be "Here are the **incomplete** tasks in your list:" ?

sanjukta99 commented on src/main/java/leduc/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,109 @@
+        else {
+            userSubstring = user.substring(8);
+        }
+        if(userSubstring.isBlank()){
```
Is the indentation for the if-else statement correct? 
Also, consider putting a blank line between the different if-else blocks to make it more readable

sanjukta99 commented on src/main/java/leduc/command/EditCommand.java:
```diff
@@ -0,0 +1,126 @@
+            DateComparisonEventException {
+        ui.display("\t Please choose the task to edit from the list by its index: ");
+        ListCommand listCommand = new ListCommand(user);
+        listCommand.execute(tasks,ui,storage);
```
Is the indentation correct for this part?

sanjukta99 commented on src/main/java/leduc/command/EditCommand.java:
```diff
@@ -0,0 +1,126 @@
+                    else {
+                        throw new MeaninglessException();
+                    }
+                }
```
Perhaps a more descriptive name for the MeaninglessException would be better?

sanjukta99 commented on src/main/java/leduc/command/EventCommand.java:
```diff
@@ -0,0 +1,113 @@
+                    priority = Integer.parseInt(prioritySplit[1].trim());
+                }
+                catch(Exception e){
+                    throw new PrioritizeLimitException();
```
Is the indentation for the try-catch correct?

sanjukta99 commented on src/main/java/leduc/command/ShortcutCommand.java:
```diff
@@ -0,0 +1,271 @@
+            setOneShortcut("list", newShortcut);
+
+            do{
+                ui.display("The precedent shortcut for help is " + HelpCommand.getHelpShortcut() +" please enter new shortcut");
```
Could all these do-while loops for each of the commands be generalized into a single do-while loop?

sanjukta99 commented on src/main/java/leduc/task/EventsTask.java:
```diff
@@ -0,0 +1,125 @@
+
+
+
+
```
Can delete all these empty lines

rshah918 committed:
Display unfinished tasks, User Guide and Developer Guide grammar fix and updates, some refactoring

sanjukta99 reviewed (commented):
Overall, good job. Please keep in mind the Java Coding Standards that can be found on the module website and the necessary changes. Take a look at my comments for some other suggestions.

laurenthg committed:
Merge pull request #86 from rshah918/master

Display Unfinished Tasks

laurenthg committed:
make text-ui-test pass

laurenthg committed:
make Junit test more unit

laurenthg committed:
update shorcut command

laurenthg committed:
update UG

laurenthg committed:
update help command

laurenthg committed:
clean code

laurenthg committed:
Merge pull request #87 from laurenthg/master

Improve Junit test, clean code, update shortcut, help, UG

laurenthg committed:
add Sequence Diagram to DG

huangje committed:
Implementation of shortcut in the DG

add 3 sequence diagram

huangje committed:
the image in DG are not shown

huangje committed:
really solved the problem for image

huangje committed:
Update [AY1920S1-CS2113-T16-1]-[Le Duc]-DG.adoc

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
Merge branch 'master' of github.com:huangje/main

rshah918 committed:
Merge pull request #88 from laurenthg/master

add Sequence Diagram to DG

rshah918 committed:
Merge pull request #4 from AY1920S1-CS2113-T16-1/master

update dg

rshah918 committed:
Merge branch 'master' into master

rshah918 committed:
Merge pull request #89 from huangje/master

Shortcut Implementation in DG

rshah918 committed:
Merge pull request #5 from AY1920S1-CS2113-T16-1/master

update fork

laurenthg committed:
DG Implementation part

huangje committed:
Merge pull request #90 from laurenthg/master

DG Implementation part

rshah918 committed:
DG Implementation section

rshah918 committed:
Merge branch 'master' into master

rshah918 committed:
Update [AY1920S1-CS2113-T16-1]-[Le Duc]-DG.adoc

laurenthg committed:
Merge pull request #91 from rshah918/master

DG Update

laurenthg committed:
update DG

rshah918 committed:
Merge pull request #92 from laurenthg/master

update DG

laurenthg committed:
jar file

huangje committed:
Merge pull request #93 from laurenthg/master

jar file

huangje committed:
Update DG

add spectific target
Change the user story to be more clear

laurenthg committed:
Merge pull request #94 from AY1920S1-CS2113-T16-1/huangje-patch-1

Update DG

rshah918 committed:
Merge pull request #6 from AY1920S1-CS2113-T16-1/master

update fork

laurenthg committed:
Move tests folder

Possibly needed to add text file to jar ?

rshah918 committed:
Change "Deadline" task to "Homework" task

rshah918 committed:
Change "Deadline" task to "Homework" task, refactor, update UG/DG

rshah918 committed:
Change "Deadline" task to "Homework" task, refactor, update UG/DG

laurenthg committed:
jar file run without to be in the correct path

laurenthg committed:
Merge pull request #96 from rshah918/master

"Homework" Task, Major Refactoring, UG/DG Update

laurenthg committed:
Make Junit test pass

laurenthg committed:
Merge branch 'jar' of github.com:laurenthg/main into debug

laurenthg committed:
make text-ui-test pass

laurenthg committed:
add jar for test without txt file

laurenthg committed:
Merge pull request #97 from laurenthg/jar

Make jar file and tests run

huangje committed:
add class diagram for the main class

huangje committed:
clean up some code

in class Date, d was change to date for more clarity

huangje committed:
correct the user guide for show and shortcut

laurenthg committed:
edit command: multi step and one-shot

laurenthg committed:
pass text-ui-test

laurenthg committed:
Junit test : use before and after

huangje committed:
add the skeleton for adding a new language

the next commit will change all the ui for the command

laurenthg committed:
Junit test for one shot edit command

laurenthg committed:
add edit to help command

laurenthg committed:
delete edit priority

laurenthg committed:
update edit in UG

laurenthg committed:
Merge pull request #98 from huangje/master

Update the user guide and the developper guide

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main into dev

huangje committed:
add new ui method for every method

huangje committed:
add new ui (different language) for exception

huangje committed:
Merge pull request #99 from laurenthg/dev

One shot edit command and before/after in Junit

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
resolve merge conflict

laurenthg committed:
Merge pull request #101 from huangje/master

Add the possibily to change the language of the program

laurenthg committed:
fix message error for EditFormatException

laurenthg committed:
update UG DG

laurenthg committed:
UD UG pdf

huangje committed:
update DG

add a new class diagram
add a sequence diagram for language

laurenthg committed:
update my PPP

huangje committed:
update the PPP

huangje committed:
Merge pull request #102 from laurenthg/dev

edit exception in Ui, DG and my PPP

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

laurenthg committed:
Merge pull request #103 from huangje/master

Update my PPP

laurenthg committed:
v1.3 jar file

huangje committed:
Merge pull request #104 from laurenthg/dev

v1.3 jar file

rshah918 committed:
fix remind bug

laurenthg committed:
v1.3 UG UD

huangje committed:
Merge pull request #106 from rshah918/master

fix remind bug

huangje committed:
Merge pull request #107 from laurenthg/dev

v1.3 UG UD

rshah918 committed:
fix remind bug

rshah918 committed:
Create PPP, update UG/DG

rshah918 committed:
Delete [AY1920S1-CS2113-T16-1]-[Le Duc]-DG.adoc

rshah918 committed:
Delete [AY1920S1-CS2113-T16-1]-[Le Duc]-UG.adoc

rshah918 committed:
Merge pull request #7 from AY1920S1-CS2113-T16-1/master

update fork

rshah918 committed:
Upgrade Statistics command

rshah918 committed:
Upgrade Statistics command

rshah918 committed:
Upgrade Statistics command

laurenthg committed:
add sort by type of task and by done or not

laurenthg committed:
update UP DG help for sort

laurenthg committed:
Junit test sort

laurenthg committed:
fix bug issue: 111

huangje committed:
Merge pull request #140 from laurenthg/sort

Add sort by done and type / Fix bug issue: 109, 122 and 133

laurenthg committed:
fix bug issue : 112 and 113

laurenthg committed:
pass junit test

laurenthg committed:
bug : issue 115

laurenthg committed:
error message in edit fix in ui

laurenthg committed:
fix issue : 127

laurenthg committed:
fix bug issue: 129

huangje committed:
add the possibility to exit the edit shortcut mode

correct the fact at when charging the file, the shortcut was not loaded
properly

huangje committed:
completed the precedent commit

huangje committed:
correct list shortcut bug

when a shortcut is attributed to list, the command list doesn't work
anymore- fixed it

huangje committed:
update message display to the user

update the message so the user can more easily understant what he has to
do

huangje committed:
Merge pull request #141 from laurenthg/eventClash

Fix event bug : issue 111,112,113

huangje committed:
Merge pull request #142 from laurenthg/editBug

Fix Edit bug: issue 115

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
the Ui.png has been updated

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main into prioritizeBug

huangje committed:
more info for the user

tell the user that his selected language is not available

huangje committed:
Merge pull request #143 from laurenthg/prioritizeBug

Fix  Prioritize bug: issue 127 and 129

laurenthg committed:
Merge pull request #144 from huangje/master

Fix bugs

rshah918 committed:
Bug Fixes, UG/DG/PPP update

rshah918 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/[AY1920S1-CS2113-T16-1]-Le Duc-DG.adoc
#	docs/[AY1920S1-CS2113-T16-1]-Le Duc-UG.adoc
#	docs/[AY1920S1-CS2113-T16-1]-Le Duc-UG.pdf
#	docs/[AY1920S1-CS2113-T16-1]-[Le Duc]-DG.pdf
#	src/main/java/leduc/command/FindCommand.java
#	src/main/java/leduc/command/StatsCommand.java

rshah918 committed:
Merge branch 'master' into master

rshah918 committed:
UG/DG fix and update

rshah918 committed:
UG/DG fix and update

rshah918 committed:
Merge remote-tracking branch 'origin/master'

laurenthg committed:
update my PPP and UG

laurenthg committed:
Merge pull request #146 from laurenthg/documentation

Update my PPP and UG

huangje committed:
add recurrence for homework

huangje committed:
update PPP

huangje committed:
update Ui

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

rshah918 committed:
Find and Statistics French Compatibility, Refactoring

rshah918 committed:
Merge pull request #147 from huangje/master

add recurrence

rshah918 committed:
Merge branch 'master' into master

rshah918 committed:
Find and Statistics French Compatibility, Refactoring

rshah918 committed:
Merge remote-tracking branch 'origin/master'

rshah918 committed:
Merge pull request #145 from rshah918/master

Upgrade Statistics feature, Fix bugs, update UG/DG/PPP

huangje committed:
add recurrence for event

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
update PPP

laurenthg committed:
Merge pull request #148 from huangje/master

add recurrence for event

huangje committed:
forget a brack, little patch

laurenthg committed:
Merge pull request #149 from huangje/master

forget a brack, little patch

rshah918 committed:
Merge pull request #8 from AY1920S1-CS2113-T16-1/master

update fork

huangje committed:
add junit test for uiFr

correct bracket in UiEn
correct Junit for event

laurenthg committed:
Merge pull request #151 from huangje/master

add junit test for uiFr

laurenthg committed:
pass text-ui-test and jar file 1.3.1

laurenthg committed:
demo duke welcome file

laurenthg committed:
test demo

huangje committed:
Merge pull request #152 from laurenthg/master

pass text-ui-test, jar file 1.3.1 and duke file for the demo

rshah918 committed:
Merge pull request #9 from AY1920S1-CS2113-T16-1/master

update fork

rshah918 committed:
Fixed Statistics bug

laurenthg committed:
Merge pull request #153 from rshah918/master

Fixed Statistics Bug

huangje committed:
add uiEnTest

correct uiFr

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

rshah918 committed:
Merge pull request #154 from huangje/master

Add Junit test for UiEn

rshah918 committed:
Fix Find Feature-limit to 5 returned matches

huangje committed:
Merge pull request #155 from rshah918/master

Fix Find Feature-limit to 5 returned matches

rshah918 committed:
Increase search acceptance threshold -50% to filter irrelevant matches

rshah918 committed:
Increase search acceptance threshold -50% to filter irrelevant matches

laurenthg committed:
Merge pull request #156 from rshah918/master

Find Feature Improvement/Bug Fix

laurenthg committed:
change [✗] to [X] and [✓] to [V]

rshah918 committed:
Merge pull request #157 from laurenthg/master

change [✗] to [X] and [✓] to [V]

rshah918 committed:
Merge pull request #10 from AY1920S1-CS2113-T16-1/master

update fork

rshah918 committed:
code cleanup

laurenthg committed:
pass Junit test

laurenthg committed:
update UG DG PPP, priority 1 to 9

laurenthg committed:
clean dead code in Ui

rshah918 committed:
PPP update, and UG update

rshah918 committed:
Generate PDF's

rshah918 committed:
DG update

rshah918 committed:
code cleanup

laurenthg committed:
fix bug reschedule event dateconflict

laurenthg committed:
Merge pull request #158 from rshah918/master

Code Cleanup, UG/DG/PPP updates

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main

huangje committed:
update Junit test

huangje committed:
Merge pull request #159 from laurenthg/master

Fix bug conflict date when edit a task / clean code /

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

laurenthg committed:
Allow generate javadoc / priority 1 to 9

huangje committed:
Merge pull request #160 from laurenthg/master

Allow generate javadoc / priority 1 to 9

huangje committed:
add missing javadoc

change user to userInput in all command for more clarity
ui, uifr, uien is in the same package

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
correct bug for recu

writting ... recu with nothing behind was correct-->fixed

huangje committed:
update the UG

laurenthg committed:
add date verification : fix #119 and #131

laurenthg committed:
Merge pull request #161 from huangje/master

update and fix

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main

laurenthg committed:
Junit Deadline -> homework

laurenthg committed:
Junit deadline -> Homework

laurenthg committed:
fix bug with keyword

huangje committed:
Merge pull request #162 from laurenthg/master

Fix bug

rshah918 committed:
Merge pull request #11 from AY1920S1-CS2113-T16-1/master

update fork

rshah918 committed:
DG added required sections.

huangje committed:
Merge pull request #163 from rshah918/master

Added required DG sections

rshah918 committed:
RemindCommand Test

rshah918 committed:
SetWelcomeMessage JUnit tests

rshah918 committed:
Change UI empty list message from "There is any tasks yet" to "There aren't any tasks!!" (grammar fix)

huangje committed:
the recu has a threshold

the nb of recurrence can't be negative, and can't exceed a certain
threshold

huangje committed:
update UG and my PPP

huangje committed:
Merge pull request #164 from rshah918/master

JUnit Tests

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

laurenthg committed:
Merge pull request #165 from huangje/master

Modify recu, update doc

laurenthg committed:
clean code

rshah918 committed:
Merge pull request #166 from laurenthg/master

clean code

rshah918 committed:
Merge pull request #12 from AY1920S1-CS2113-T16-1/master

update fork

rshah918 committed:
fix RemindCommandTest

laurenthg committed:
Merge pull request #167 from rshah918/master

Fix RemindCommandTest

huangje committed:
update my PPP

laurenthg committed:
Merge pull request #168 from huangje/master

update my PPP

laurenthg committed:
UG DG and PPP

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main into documentation

huangje committed:
Merge pull request #169 from laurenthg/documentation

Documentation UG DG PPP

rshah918 committed:
Merge pull request #13 from AY1920S1-CS2113-T16-1/master

update fork

rshah918 committed:
Instructions for Manual Testing sections 1-5

rshah918 committed:
PPP Update

huangje committed:
Merge pull request #170 from rshah918/master

Instructions for manual testing sections 1-5

rshah918 committed:
Appendix G5: Find Command

rshah918 committed:
Complete "Instructions for manual testing" parts 1-11

huangje committed:
update PPP/UG/DG

rshah918 committed:
Generate PDF's, DG typo fixes

huangje committed:
add the pdf

huangje committed:
Merge pull request #171 from rshah918/master

Instructions for Manual testing sections G.1-G.11

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

huangje committed:
change format DG

rshah918 committed:
Merge pull request #172 from huangje/master

Update PPP/UD/GD

laurenthg committed:
update DG(manuel testing ) and PPP

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main

laurenthg committed:
Merge pull request #173 from laurenthg/master

update DG(manuel testing ) and PPP

laurenthg committed:
prepare shadowjar to generate jar

huangje committed:
update name of UG/DG/PPP

huangje committed:
Merge github.com:AY1920S1-CS2113-T16-1/main

laurenthg committed:
initial welcome message in file created by Jar

huangje committed:
update name

rshah918 committed:
Merge pull request #174 from huangje/master

Update name

laurenthg committed:
create data file with tasks when executing jar file

laurenthg committed:
update ppp

laurenthg committed:
add jar file

laurenthg committed:
Merge https://github.com/AY1920S1-CS2113-T16-1/main

rshah918 committed:
Merge pull request #175 from laurenthg/master

Jar file v1.4

huangje committed:
correct ug

rshah918 committed:
Merge pull request #176 from huangje/master

correct ug
